docs/source/multiple-threads.rst

```diff
@@ -0,0 +1,150 @@
+Use from Multiple Threads
+#########################
+
+This document describes how various Kokkos functionality can safely be called when using more than one thread of host execution.
+It also describes intentions for what Kokkos aspires to implement in support of known or anticipated use cases, with clear indications that they are not currently supported.
+Some discussion of implementation characteristics and performance expectations is included as well.
+
+We will use terms as defined by the C++ standard memory model. In that model, executions that include a data race have undefined behavior.
+Thus, use of Kokkos in ways that don't conform to the safety guidance given here may be considered to result in undefined behavior.
+
+In formal terms, we address two distinct issues here.
+*Concurrency* is the matter of what operations may safely execute in different threads. 
+*Linearizability* is the matter of how the results of concurrently-executed operations may map to potential observed orders in a hypothetical non-concurrent execution.
+
+Core
+====
+
+Fundamental Operations
+----------------------
+
+For purposes of multi-threaded execution, all free and member functions provided by Kokkos can be thought of as performing a sequence of one or more of the following *fundamental operations*
+
+
+* *Initialization*
+
+.. Not just Kokkos::init, but also whatever device-specific or thread-specific stuff we have Legion doing now
+
+* *Finalization*
+
+.. Ditto Initialization
```

Comment:
Backends can still only be initialized or finalized once. I'm not quite sure if it's worth mentioning initialization/finalization then. At the very least, we need to clarify what we mean here (execution space instance initialization/finalization maybe sensible).